app/src/main/java/com/fpvout/digiview/vr/views/VrView.java

```diff
@@ -0,0 +1,190 @@
+import android.view.LayoutInflater;
+import android.view.Surface;
+import android.view.SurfaceHolder;
+import android.view.SurfaceView;
+import android.view.View;
+import android.widget.FrameLayout;
+
+import androidx.annotation.NonNull;
+import androidx.constraintlayout.widget.ConstraintLayout;
+
+import com.fpvout.digiview.R;
+import com.fpvout.digiview.VideoReaderExoplayer;
+import com.fpvout.digiview.vr.gles.EglCore;
+import com.fpvout.digiview.vr.gles.FullFrameRect;
+import com.fpvout.digiview.vr.gles.Texture2dProgram;
+import com.fpvout.digiview.vr.gles.WindowSurface;
+import com.google.android.exoplayer2.SimpleExoPlayer;
+
+public class VrView extends FrameLayout implements SurfaceTexture.OnFrameAvailableListener {
+    private SurfaceView surfaceViewLeft;
+    private SurfaceView surfaceViewRight;
+    private SimpleExoPlayer mPlayer;
+    private EglCore eglCore;
+    private FullFrameRect fullFrameBlit;
+    private int textureId = 0;
+    private SurfaceTexture videoSurfaceTexture;
+    private final float[] transformMatrix = new float[16];
+    private WindowSurface mainDisplaySurface;
+    private WindowSurface secondaryDisplaySurface;
+    private Surface surface;
+
+    public VrView(Context context) {
+        super(context);
+        init(context);
+    }
+
+    public VrView(Context context, AttributeSet attrs) {
+        super(context, attrs);
+        init(context);
+    }
+
+    public VrView(Context context, AttributeSet attrs, int defStyle) {
+        super(context, attrs, defStyle);
+        init(context);
+    }
+
+    private void init(Context context) {
+        LayoutInflater inflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
+        inflater.inflate(R.layout.vr_view, this);
+        surfaceViewLeft = findViewById(R.id.surfaceViewLeft);
+        surfaceViewRight = findViewById(R.id.surfaceViewRight);
+    }
+
+    public void setAspect(int dividend, int divisor) {
+        ConstraintLayout.LayoutParams leftParams = (ConstraintLayout.LayoutParams) surfaceViewLeft.getLayoutParams();
+        ConstraintLayout.LayoutParams rightParams = (ConstraintLayout.LayoutParams) surfaceViewRight.getLayoutParams();
+        String dimensionString = "H," + dividend + ":" + divisor;
+        leftParams.dimensionRatio = dimensionString;
+        rightParams.dimensionRatio = dimensionString;
+    }
+
+    public void start(SimpleExoPlayer mPlayer) {
+        this.mPlayer = mPlayer;
+        surfaceViewLeft.getHolder().addCallback(videoSurfaceCallbackLeft);
+        surfaceViewRight.getHolder().addCallback(videoSurfaceCallbackRight);
+        surfaceViewLeft.setVisibility(View.GONE);
+        surfaceViewRight.setVisibility(View.GONE);
+        surfaceViewLeft.post(new Runnable() {
+            @Override
+            public void run() {
+                surfaceViewLeft.setVisibility(View.VISIBLE);
+                surfaceViewRight.setVisibility(View.VISIBLE);
+            }
+        });
```

Comment:
Nice work @0TheRain0 , didn't know about the `Grafika` repo, really a hidden gem.
Your `VrView` works well, though I don't have a phone holder with lenses to test.

What is the idea behind this code? To retrigger the `surfaceCreated`?
If we don't set to `GONE` first,  could it trigger the callback already before the view is set to GONE?

I'm just registering on  view inflation, and expect the `surfaceView` to be later ready, while this is a racecondition, in practice it works well for me and it inverts the dependency, so that the `VrView` doesn't have to know about the player.

https://github.com/d4rken/fpv-dvca/commit/06d7565c55251ca48ead0da65ff9eaa593790c52#diff-c3426db1c549d5574f8210d209aa690c2fda7d05d83b3f98548ea81e7b3fca8b

Reply:
@d4rken if someone else would like to take ownership of this code change and make the adjustments needed I would appreciate it.  I no longer have time to work on it.